### ISSUE TITLE
Support relative urls in index.yaml generated from ChartMuseum and Azure Container Registry Helm

### ISSUE THREAD

**Issue by vazexqi** [spinnaker/spinnaker#4078]
### Issue Summary:

[IndexParser](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/helm/IndexParser.java) seems to only understand full URLS in Helm repository's index.yaml.

### Cloud Provider(s):

Any

### Environment:

Any

### Feature Area:

CloudDriver

### Description:

Many of the popular helm repositories use full urls in their index.yaml. For instance, https://azure-samples.github.io/helm-charts/index.yaml and https://kubernetes-charts.storage.googleapis.com/.

However, some other repositories can generate relative paths. For instance the ones from [Azure Container Registry Helm Repos](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-helm-repos) and also from [ChartMuseum](https://github.com/helm/chartmuseum). 

*Azure*

```
apiVersion: v1
entries:
  wordpress:
  - acrMetadata:
      manifestDigest: sha256:08ef434162070bba4256414c80b001d15b7503ef2a1a4fa1f60bab174f80d4d7
    appVersion: 5.1.0
    created: "2019-03-06T16:59:25.8892193Z"
    description: Web publishing platform for building blogs and websites.
    digest: e0f315e609e02fd4798bed33d56267fd68f5e45dae2015b8895474178e6cc8e4
    engine: gotpl
    home: http://www.wordpress.com/
    icon: https://bitnami.com/assets/stacks/wordpress/img/wordpress-stack-220x234.png
    keywords:
    - wordpress
    - cms
    - blog
    - http
    - web
    - application
    - php
    maintainers:
    - email: containers@bitnami.com
      name: Bitnami
    name: wordpress
    sources:
    - https://github.com/bitnami/bitnami-docker-wordpress
    urls:
    - _blobs/wordpress-5.2.5.tgz
    version: 5.2.5
generated: "2019-03-06T18:44:57Z"
serverInfo:
  contextPath: /helm/v1/repo
```

*ChartMuseum*

```
apiVersion: v1
entries:
  wordpress:
  - apiVersion: v1
    appVersion: 5.1.0
    created: "2019-03-06T15:49:19.755782306-08:00"
    description: Web publishing platform for building blogs and websites.
    digest: 7f068075264b5ba7c14c23672698152ae6a3ebac1c47916e4efe19cd624d5fce
    engine: gotpl
    home: http://www.wordpress.com/
    icon: https://bitnami.com/assets/stacks/wordpress/img/wordpress-stack-220x234.png
    keywords:
    - wordpress
    - cms
    - blog
    - http
    - web
    - application
    - php
    maintainers:
    - email: containers@bitnami.com
      name: Bitnami
    name: wordpress
    sources:
    - https://github.com/bitnami/bitnami-docker-wordpress
    urls:
    - charts/wordpress-5.4.0.tgz
    version: 5.4.0
generated: "2019-03-06T15:49:28-08:00"
serverInfo: {}

```

I *suspect* this is what is causing this issue based on the error below and also the fact that if I switch to a different repository with full URLs, it works. The way that [IndexParser](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/helm/IndexParser.java) works is just to extract and use the url as-is. 

I can't find the spec for index.yaml but helm CLI works fine with both of the formats above.

### Steps to Reproduce:

1. Use one of the helm repositories that do not put full urls.
2. Try to create a new Bake (Manifest) stage
3. Run the pipeline and notice the following error:

```
2019-03-07 01:35:52.719  WARN 1 --- [      MvcAsync8] c.n.s.c.a.helm.HelmArtifactCredentials   : Invalid url: 
2019-03-07 01:35:52.723 ERROR 1 --- [0.0-7002-exec-9] c.n.s.k.w.e.GenericExceptionHandlers     : Internal Server Error

com.netflix.spinnaker.clouddriver.artifacts.helm.HelmArtifactCredentials$FailedDownloadException: Unable to download the contents of artifact
        at com.netflix.spinnaker.clouddriver.artifacts.helm.HelmArtifactCredentials.download(HelmArtifactCredentials.java:113) ~[clouddriver-artifacts.jar:na]
        at com.netflix.spinnaker.clouddriver.artifacts.ArtifactDownloader.download(ArtifactDownloader.java:62) ~[clouddriver-artifacts.jar:na]
        at com.netflix.spinnaker.clouddriver.controllers.ArtifactController.lambda$fetch$0(ArtifactController.java:65) ~[clouddriver-web.jar:na]
        at org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBodyReturnValueHandler$StreamingResponseBodyTask.call(StreamingResponseBodyReturnValueHandler.java:106) ~[spring-webmvc-4.3.20.RELEASE.jar:4.3.20.RELEASE]
        at org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBodyReturnValueHandler$StreamingResponseBodyTask.call(StreamingResponseBodyReturnValueHandler.java:93) ~[spring-webmvc-4.3.20.RELEASE.jar:4.3.20.RELEASE]
        at org.springframework.web.context.request.async.WebAsyncManager$5.run(WebAsyncManager.java:332) ~[spring-web-4.3.20.RELEASE.jar:4.3.20.RELEASE]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_191]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_191]
        at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_191]
```

**Comment by ezimanyi** [comment]
Closed by spinnaker/clouddriver#3515, to be in Spinnaker 1.14.  Thanks @dongo0223!

**Comment by nitishm** [comment]
Was this verified that it works? I am seeing a similar issue when fetching the helm chart artifact from artifactory in 1.17.10. 
```
    helm:
    enabled: true
    accounts:
    - name: my-artifactory
      repository: http://10.15.x.x:8082/artifactory/cnhelm
      username: nitish_malhotra 
      password: xxxxxx
```
